### PR TITLE
Add latency metrics support

### PR DESCRIPTION
**Pull request by LorcanMcVeigh**
### Proposed changes
Add support for the Prometheus latency metric.  

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/nginx-ingress-operator/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

File: pkg/controller/nginxingresscontroller/utils.go [hunk 1]
```diff
@@ -103,6 +103,10 @@ func generatePodArgs(instance *k8sv1alpha1.NginxIngressController) []string {
 		if instance.Spec.Prometheus.Port != nil {
 			args = append(args, fmt.Sprintf("-prometheus-metrics-listen-port=%v", *instance.Spec.Prometheus.Port))
 		}
+
+		if instance.Spec.EnableLatencyMetrics {
```
Comment (Dean-Coakley) [inline pkg/controller/nginxingresscontroller/utils.go:107]: Can you add test coverage in `pkg/controller/nginxingresscontroller/utils_test.go` ?

**Review by Dean-Coakley — APPROVED** [review]

File: deploy/crds/k8s.nginx.org_nginxingresscontrollers_crd.yaml [hunk 2]
```diff
@@ -181,6 +181,11 @@ spec:
               required:
               - enable
               type: object
+            enableLatencyMetrics:
```
Comment (pleshakov) [inline deploy/crds/k8s.nginx.org_nginxingresscontrollers_crd.yaml:184]: I believe this file is autogenerated. So it should have the description  Enable collection of latency metrics for upstreams" taken from pkg/apis/k8s/v1alpha1/nginxingresscontroller_types.go

**Review by pleshakov — COMMENTED** [review]
Hi @LorcanMcVeigh 

please see my comments

**Review by pleshakov — APPROVED** [review]
🚀

**Pull request merged**